lib/httplib/httplib.go

```diff
@@ -150,7 +150,7 @@ func ReadJSON(r *http.Request, val interface{}) error {
 func ConvertResponse(re *roundtrip.Response, err error) (*roundtrip.Response, error) {
 	if err != nil {
 		if uerr, ok := err.(*url.Error); ok && uerr != nil && uerr.Err != nil {
```

Comment:
use `errors.As`

```go
func ConvertResponse(re *roundtrip.Response, err error) (*roundtrip.Response, error) {
	if err != nil {
	        var uErr *url.Error
		if errors.As(err, &uErr) && uErr.Err != nil {

```

Reply:
url.Error has `Temporary` and `Timeout` methods that might be useful to change our error types as well.

Reply:
Thanks. the docs for Temporary say 'Do not use this method' which worry me. 
I've implemented `errors.As` and must admit, it's much cleaner!